sdk/test/opentelemetry/sdk/trace/span_test.rb

```diff
@@ -60,10 +60,10 @@
       _(span.attributes).must_equal('foo' => 'bar')
     end
 
-    it 'trims the oldest attribute' do
+    it 'trims the newest attribute' do
```

Comment:
I believe this also happens during construction. Bonus points for adding test coverage there.

Reply:
We have a test, but it only asserts the length rather than the contents: https://github.com/open-telemetry/opentelemetry-ruby/blob/dcda280c9e919d66038a452b29c268434a4138c6/sdk/test/opentelemetry/sdk/trace/span_test.rb#L501-L508
I'll add an assertion for the contents.

Reply:
https://github.com/open-telemetry/opentelemetry-ruby/pull/1536/commits/ba0eacca7512f2b3cd424a6e2dcb092fcd1672e2